tests/draft2020-12/optional/format/email.json

```diff
@@ -47,6 +72,16 @@
                 "description": "two subsequent dots inside local part are not valid",
                 "data": "te..st@example.com",
                 "valid": false
+            },
+            {
+                "description": "an invalid domain",
+                "data": "joe.bloggs@invalid=domain.com",
+                "valid": false
+            },
+            {
+                "description": "an invalid IPv4-address-literal",
+                "data": "joe.bloggs@[127.0.0.300]",
+                "valid": false
```

Comment:
It might be a bit much to expect parsers to extract the IP address and then perform a separate IPv4 validation on it.

Reply:
Doesn't have to be done as a separate step, a grammar can express what is and is not a valid IPv4 address, as well as a valid IPv6 address. See https://github.com/gene-hightower/smtp-address-parser for TypeScript, see https://github.com/gene-hightower/smtp-address-validator for Ragel/C++ code. In each case the rule for "snum" is a bit more involved than the syntax in RFC-5321, but it does the job.

Reply:
Although it would be quite large, this can even be expressed as a regular expression.
Regardless of how it's done, it's still something I would expect out of email address validation, I think the test is fair.